config/config-api/src/com/thoughtworks/go/config/BasicCruiseConfig.java

```diff
@@ -1356,6 +1216,16 @@ public boolean canDeletePluggableSCMMaterial(SCM scmConfig) {
     }
 
     @Override
+    public void setPartials(List<PartialConfig> partials) {
+        this.partials = new Cloner().deepClone(partials);
```

Comment:
I'd consider doing this outside and forcing the caller to make this decision. For instance, maybe this is already cloned (it's not, today). I'd think that [this](https://github.com/gocd/gocd/pull/2256/files#diff-e236b0af07a149fc225404e31015c033R256) is a better place to clone. It's explicit there that it's an attempt at saving. So, you're saying you need it to throw away the clone if it does not work.